test/simulation.jl

```diff
@@ -0,0 +1,200 @@
+using WangLandau
+using Test
+using Random
+using CommonSolve
+using Logging 
+
+include(joinpath(@__DIR__, "..", "examples", "ising.jl"))
+
+@testset "WangLandauSimulation" begin
+    Random.seed!(12345)
+
+    L = 5
+    periodic = false
+    statedefn = Ising2D(L; periodic)
+
+    sim = WangLandau.WangLandauSimulation(statedefn)
+    @test isa(sim, WangLandau.WangLandauSimulation)
+    @test isa(sim.logf_strategy, WangLandau.ReduceByFactor)
+    @test isa(sim.flat_strategy, WangLandau.FractionOfMean)           
+    @test isa(sim.catchup_strategy, WangLandau.NoCatchup)    
```

Comment:
Here and elsewhere, you don't need `WangLandau.Thing` for exported objects. Check the module file `WangLandau.jl` for the list.
```suggestion
    sim = WangLandauSimulation(statedefn)
    @test isa(sim, WangLandauSimulation)
    @test isa(sim.logf_strategy, ReduceByFactor)
    @test isa(sim.flat_strategy, FractionOfMean)           
    @test isa(sim.catchup_strategy, NoCatchup)    
```